pkg/icingadb/ha.go

```diff
@@ -78,10 +81,18 @@ func (h *HA) Done() <-chan struct{} {
 	return h.done
 }
 
+// Environment returns the current environment.
+func (h *HA) Environment() *v1.Environment {
+	h.environmentMu.Lock()
+	defer h.environmentMu.Unlock()
+
+	return &h.environment
+}
```

Comment:
Locking here doesn't solve the issue. The controller loop might still update `h.environment.Id` after the pointer was returned here. Options would be:
1. Return a copy here (taking the copy is synchronized, afterwards they're independent objects, so no more need to sync)
2. Make `h.environment` a `*Environment` and never update existing `Environment` instances but instead always create a new one (after initialization, all accesses are read-only and need no further synchronization)
3. Expose `h.environmentMu` to allow the caller to properly synchronize (wouldn't recommend doing this but would theretically be possible)

Reply:
Implemented option 2, thanks!